### PR TITLE
Minor Performances Improvement:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Unnecessary conversion to String
* Use Final
* Use StandarCharset
* Unnecessary conversion to String
* Unnecessary enum modifier
* Unnecessary interface modifier
* Unnecessary semicolon
* Unused import
* Unnecessary boxing
* Unnecessary unboxing
* Collapse catch
* Use Lambda
* Use StringBuilder

### CODE REVIEW

**Comment by arturobernalg** [comment]
Sorry Guys, I know are to many changes and to many file. But this will be improve the code and make more readable, and in some case and improvement

**Comment by CesarCoelho** [comment]
@arturobernalg - Could you let us know what is the process that you are following to do these changes?
For example, are you getting the suggestions from some tool, or?

**Comment by arturobernalg** [comment]
Hi @CesarCoelho 
some of them are from PMD check.  Other checking the code. And other god practice of java

**Comment by arturobernalg** [comment]
The idea:
try to make the code  as most simple posible, making all the method  read-only.
Intent to make all objects immutable.
prevent a developer from modifying an object by mistake

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/consumer/ArchiveSyncGenAdapter.java [hunk 1]
```diff
@@ -41,9 +41,9 @@
     private final Map<Integer, byte[]> receivedChunks;
     private final Semaphore completed;
 //    private Long interactionTicket = null;
-    private UInteger numberOfChunks = null;
-    private long lastTimeReceived = 0;
-    private long lastknowIndex = 0;
+    private UInteger numberOfChunks;
+    private long lastTimeReceived;
+    private long lastknowIndex;
```
Comment (CesarCoelho) [inline core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/consumer/ArchiveSyncGenAdapter.java:46]: These changes make the code less readable. The zeros make it explicit that the initialization IS zero.

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/ArchiveProviderServiceImpl.java [hunk 2]
```diff
@@ -62,8 +62,8 @@
 
     private final ArchiveManager manager = new ArchiveManager(null);
     private MALProvider archiveServiceProvider;
-    private boolean initialiased = false;
-    private boolean running = false;
+    private boolean initialiased;
+    private boolean running;
```
Comment (CesarCoelho) [inline core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/ArchiveProviderServiceImpl.java:66]: Same here. I prefer to have explicit initializations as it doesn't leave room for ambiguity.

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/EventProviderServiceImpl.java [hunk 3]
```diff
@@ -72,10 +72,10 @@
 public class EventProviderServiceImpl extends EventInheritanceSkeleton {
 
     private MALProvider eventServiceProvider;
-    private boolean initialiased = false;
-    private boolean running = false;
+    private boolean initialiased;
+    private boolean running;
     private final Object lock = new Object();
-    private boolean isRegistered = false;
+    private boolean isRegistered;
```
Comment (CesarCoelho) [inline core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/EventProviderServiceImpl.java:78]: Same comment -> Explicit is better

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/util/ObjectInstanceIdGenerator.java [hunk 4]
```diff
@@ -37,7 +37,7 @@
   private final Random RNG = new Random(System.nanoTime());
   private final AtomicInteger objectInstanceIdCounter = new AtomicInteger(0);
 
-  private long lastObjectInstanceIdTimeStamp = 0;
+  private long lastObjectInstanceIdTimeStamp;
```
Comment (CesarCoelho) [inline core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/util/ObjectInstanceIdGenerator.java:40]: Same comment -> Explicit is better

File: core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/reconfigurable/service/PersistLatestServiceConfigurationAdapter.java [hunk 5]
```diff
@@ -50,7 +50,7 @@
     private final ArchiveInheritanceSkeleton archiveService;
     private final Long serviceConfigObjId;
     private final ExecutorService executor;
-    private Long configObjectsObjId = null;
+    private Long configObjectsObjId;
```
Comment (CesarCoelho) [inline core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/reconfigurable/service/PersistLatestServiceConfigurationAdapter.java:53]: Same comment -> Explicit is better

**Comment by CesarCoelho** [comment]
@arturobernalg : Thank you very much for your commit, we really do appreciate it!  :)
However, this kind of Big Bang Commit makes the reviews quite hard, can you submit individual Pull Requests?
It is better to take one at the time than all at once. Makes it easier for us to review and merge.

**Comment by arturobernalg** [comment]
HI @CesarCoelho 
Great. I will do that.

**Pull request closed without merging**